libraries/AP_HAL_ChibiOS/hwdef/ZeroOneX6/hwdef.dat

```diff
@@ -0,0 +1,335 @@
+# hw definition file for processing by chibios_hwdef.py
+# for the ZeroOne X6 hardware
+
+# default to all pins low to avoid ESD issues
+DEFAULTGPIO OUTPUT LOW PULLDOWN
+
+# MCU class and specific type
+MCU STM32H7xx STM32H743xx
+
+# crystal frequency
+OSCILLATOR_HZ 16000000
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 2
+
+# board ID for firmware load
+APJ_BOARD_ID 5600
+
+FLASH_RESERVE_START_KB 128
+
+# to be compatible with the px4 bootloader we need
+# to use a different RAM_MAP
+env USE_ALT_RAM_MAP 1
+
+# flash size
+FLASH_SIZE_KB 2048
+
+# supports upto 8MBits/s
+CANFD_SUPPORTED 8
+
+env OPTIMIZE -O2
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 UART7 UART5 USART1 UART8 USART2 UART4 USART3 OTG2
+
+# USB
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+PA9 VBUS INPUT OPENDRAIN
+
+# pins for SWD debugging
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# telem1
+PE8 UART7_TX UART7
+PF6 UART7_RX UART7
+PF8 UART7_RTS UART7
+PE10 UART7_CTS UART7
+
+# telem2
+PC8 UART5_RTS UART5
+PC9 UART5_CTS UART5
+PC12 UART5_TX UART5
+PD2 UART5_RX UART5
+
+# telem3
+PA3 USART2_RX USART2
+PD5 USART2_TX USART2
+PD3 USART2_CTS USART2
+PD4 USART2_RTS USART2
+define DEFAULT_SERIAL5_PROTOCOL SerialProtocol_MAVLink2
+
+# GPS1
+PB6 USART1_TX USART1
+PB7 USART1_RX USART1
+
+# GPS2
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8
+
+# uart4
+PH13 UART4_TX UART4
+PH14 UART4_RX UART4
+
+# debug uart

```

Comment:
I haven't looked at the actual DMA output, but my guess is that there is a lot of sharing given the number of SPI buses and UARTs - I would think carefully about removing DMA from some of the UARTs if at all possible

Reply:
virtually all the uart TX are shared with a spi bus...